extra/resources/ifspeed

```diff
@@ -409,20 +449,28 @@ bond_get_speed() {
         bond_get_active_iface ${iface} active_iface
         aggregate_speed=$( iface_get_speed $active_iface )
     fi
-    if [ -n "$2" ] ; then # Record value in specified var. This expects we were called not in a sub-shell. 
+    if [ -n "$2" ] ; then # Record value in specified var. This expects we were called not in a sub-shell.
         eval $2=\${aggregate_speed}
     else # Expect sub-shell
         echo ${aggregate_speed}
     fi
 }
 
 update() {
-    local speed=$( iface_get_speed ${OCF_RESKEY_iface} )
+    local nic=${OCF_RESKEY_iface};
+    if [ -z "${OCF_RESKEY_iface}" ]; then
+        nic=`get_nic_name_by_ip`
```

Comment:
could you please change this to nic=$( get_nic_name_by_ip )
just for consistency in style.

Reply:
Also, for the performance (removing need in extra subshell), you can use eval inside your function and pass name of the variable to set.
Example is bond_get_active_iface() function:
It receives a name of the variable to set in the second positional parameter and then does
<pre>eval $2=\${active}</pre> - sets that variable
It is called from bond_get_speed() the next way:
<pre>bond_get_active_iface ${iface} active_iface</pre>
Thus, in bond_get_speed(), variable 'active_iface' is set.